cylc/flow/task_job_mgr.py

```diff
@@ -985,11 +985,13 @@ def _set_retry_timers(
 
     def _simulation_submit_task_jobs(self, itasks, workflow):
         """Simulation mode task jobs submission."""
+        sim_platform = get_platform()
+        sim_platform['name'] = 'SIMULATION'
         for itask in itasks:
+            itask.platform = sim_platform
             itask.waiting_on_job_prep = False
             itask.submit_num += 1
             self._set_retry_timers(itask)
-            itask.platform = {'name': 'SIMULATION'}
```

Comment:
If this is all that's needed for simulation mode, I don't think we need to run `get_platform` at all?

Reply:
Yes.